integrationtests/all-embedded-query-it/src/test/java/org/infinispan/all/embeddedquery/QueryDslConditionsTest.java

```diff
@@ -2182,6 +2188,126 @@ public void testMissingParam() throws Exception {
       q.list();
    }
 
+   @Test
+   public void testNotIndexedProjection() throws Exception {
+      QueryFactory qf = getQueryFactory();
+
+      Query q = qf.from(getModelFactory().getTransactionImplClass())
+            .select("id", "isValid")
+            .having("id").gte(98)
+            .toBuilder()
+            .orderBy("id")
+            .build();
+      List<Object[]> list = q.list();
+
+      assertEquals(2, list.size());
+      assertEquals(2, list.get(0).length);
+      assertEquals(98, list.get(0)[0]);
+      assertEquals(true, list.get(0)[1]);
+      assertEquals(2, list.get(1).length);
+      assertEquals(99, list.get(1)[0]);
+      assertEquals(true, list.get(1)[1]);
```

Comment:
This should also work:

      assertEquals(asList(asList(98, true), asList(99, true)), list);